api_backend_server/app.py

```diff
@@ -1,25 +1,160 @@
+import datetime
+from functools import wraps
+import os
+from pathlib import Path
 
 
-from flask import Flask
-from flask import request
-from flask import jsonify
+from flask import Flask, request, jsonify, make_response
+from flask_sqlalchemy import SQLAlchemy
+from werkzeug.security import generate_password_hash, check_password_hash
+import uuid
+import jwt
 
 
 app = Flask(__name__)
 
+app.config["SECRET_KEY"] = os.getenv("SECRET_KEY", "very_very_secret_secret")
+app.config["SQLALCHEMY_DATABASE_URI"] = os.getenv(
+                                                   "SQLALCHEMY_DATABASE_URI",
+                                                   "".join([
+                                                            "sqlite:////",
+                                                            str(Path.cwd().parent.joinpath("local.db"))
+                                                            ])
+                                                  )
+app.config["SQLALCHAMY_TRACK_MODIFICATIONS"] = True
+
+
+db = SQLAlchemy(app)
+
+
+class Users(db.Model):
+    id = db.Column(db.Integer, primary_key=True)
+    public_id = db.Column(db.Integer)
+    name = db.Column(db.String(50))
+    password = db.Column(db.String(50))
+    admin = db.Column(db.Boolean)
+
+
+class Activities(db.Model):
+    id = db.Column(db.Integer, primary_key=True)
+    user_id = db.Column(db.Integer)
+    name = db.Column(db.String(50), unique=False, nullable=False)
+
+
+def token_required(f):
+    @wraps(f)
+    def wrapper(*args, **kwargs):
+
+        token = None
+
+        if "x-access-tokens" in request.headers:
+            token = request.headers["x-access-tokens"]
+
+        if not token:
+            return jsonify({"message": "a valid token is missing"})
+
+        try:
+            data = jwt.decode(token, app.config["SECRET_KEY"])
+        except jwt.exceptions.DecodeError:
+            return jsonify({"message": "token is invalid"})
+        else:
+            current_user = Users.query.filter_by(public_id=data['public_id']).first()
+
+        return f(current_user, *args, **kwargs)
+
+    return wrapper
+
 
 @app.route("/", methods=["GET"])
 def index():
     return jsonify(message="Hello from Python backend server")
 
 
-@app.route("/login", methods=["POST"])
-def login():
-    username = request.args.get("username")
-    user_id = randint(0, 10)
-    return jsonify(username=username,
-                   userid=user_id)
+@app.route("/api/register", methods=["POST"])
+def signup_user():
+    data = request.get_json()
+
+    hashed_password = generate_password_hash(data["password"], method="sha256")
+    public_id = str(uuid.uuid4())
+
+    new_user = Users(public_id=public_id, name=data["name"], password=hashed_password, admin=False)
+    db.session.add(new_user)
+    db.session.commit()
+
+    return jsonify({"message": "registered successfuly",
+                    "user_id": public_id})
+
+
+@app.route("/api/login", methods=["POST"])
```

Comment:
Analogicznie dodanie **users**:
```suggestion
@app.route("/api/users/login", methods=["POST"])
```